src/main/java/ch/njol/skript/events/EvtVehicleCollision.java

```diff
@@ -0,0 +1,89 @@
+package ch.njol.skript.events;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.entity.EntityData;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import org.bukkit.block.Block;
+import org.bukkit.block.data.BlockData;
+import org.bukkit.event.Event;
+import org.bukkit.event.vehicle.VehicleBlockCollisionEvent;
+import org.bukkit.event.vehicle.VehicleCollisionEvent;
+import org.bukkit.event.vehicle.VehicleEntityCollisionEvent;
+import org.jetbrains.annotations.Nullable;
+
+public class EvtVehicleCollision extends SkriptEvent {
+
+	static {
+		Skript.registerEvent("Vehicle Collision", EvtVehicleCollision.class, new Class[]{VehicleBlockCollisionEvent.class, VehicleEntityCollisionEvent.class}
+			,"vehicle collision [(with|of) [a[n]] %-itemtypes/blockdatas/entitydatas%]",
+			"vehicle block collision [(with|of) [a[n]] %-itemtypes/blockdatas%]",
+			"vehicle entity collision [(with|of) [a[n]] %-entitydatas%]")
+				.description("Called when a vehicle collides with a block or entity.")
+				.examples("on vehicle collision:", "on vehicle collision with obsidian:", "on vehicle collision with a zombie:")
+				.since("INSERT VERSION");
+	}
+
+	private Literal<?> expr;
+	private boolean blockCollision;
+	private boolean entityCollision;
+
+	@Override
+	public boolean init(Literal<?>[] args, int matchedPattern, ParseResult parseResult) {
+		expr = args[0];
```

Comment:
this should be evaluated during parsing
then you don't need the block/entity patterns, you know what you're looking for immediately.

Reply:
Im not sure wym? Are you talking about `vehicle block collision`  and `vehicle entity collision`?
I added those so it would be specific
`vehicle collision` is any collision for both block and entity
`vehicle block collision` is any block collision
`vehicle entity collision` is any  entity collision

Reply:
Yes i'm saying you can just evaluate what the user put in and know whether it's a block or entity or both

Reply:
Right, I understand that. But then that's dependent on them wanting to only call for a specific entity or specific block. Those are for any entity or any block
`on vehicle block collision` could be bedrock, obsidian, dirt,  etc
`on vehicle [block] collision of obsidian` could only be obsidian
Not unless I'm missing something

Reply:
ah i see why
yeah you can keep those as pattern but it should still be eval-ed in init